### DISCUSSION TITLE
Disabled checked checkboxes not rendering correctly

### DISCUSSION THREAD

**Discussion by antonsamper** [react-hook-form/react-hook-form#8049] [Q&A] [answered]
Hi there,

Is there a reason why a disabled checkbox wouldnt be rendering as checked?

See the codesandbox - It works as expected when the checked option is the last one to be rendered (options2) but not when the checked option is in the middle (options1). The code is the same for both.

https://codesandbox.io/s/nifty-shaw-yn3pg9?file=/src/App.tsx

**Comment by Moshyfawn** [comment, marked as answer]
It looks like a bug. Track it's progress in #8050

**Reply by Moshyfawn** [reply]
I should mention, that `register` expects `disabled` value to be passed into the second options argument instead of to the input element to determine which values RHF should omit + their validation
```diff
<input
  {...register("options", {
+  disabled: option.disabled
  })}
-  disabled={option.disabled}
/>
``` 
Learn more [at RHF docs](https://react-hook-form.com/api/useform/register)

**Reply by antonsamper** [reply]
thanks for creating an issue @Moshyfawn
